pkg/testing/chains.go

```diff
@@ -854,25 +860,38 @@ func (chain *Chain) QueryProof(counterparty *Chain, counterpartyClientID string,
 }
 
 func (counterparty *Chain) QueryClientProof(chain *Chain, counterpartyClientID string, height *big.Int) ([]byte, *Proof, error) {
-	cs, found, err := counterparty.IBCHost.GetClientState(
-		counterparty.CallOpts(context.Background(), RelayerKeyIndex),
-		counterpartyClientID,
-	)
-	if err != nil {
-		return nil, nil, err
-	} else if !found {
-		return nil, nil, fmt.Errorf("client not found: %v", counterpartyClientID)
-	}
 	proof, err := counterparty.QueryProof(chain, counterpartyClientID, commitment.ClientStateCommitmentSlot(counterpartyClientID), height)
 	if err != nil {
 		return nil, nil, err
 	}
 	switch counterparty.ClientType() {
 	case ibcclient.MockClient:
-		h := sha256.Sum256(cs)
+		cs := counterparty.GetMockClientState(counterpartyClientID)
```

Comment:
currently we can call GetClientState of the handler directly instead of using each client accessor